pkg/oci/oci.go

```diff
@@ -892,50 +887,45 @@ func ensureImage(ctx context.Context, imageStore oras.Target, image string, imgO
 		}
 	}
 
-	if len(imgOpts.AllowedRegistries) > 0 {
-		targetImage, err := normalizeImageName(image)
-		if err != nil {
-			return fmt.Errorf("normalizing image: %w", err)
-		}
-
-		image := targetImage.String()
+	if len(imgOpts.AllowedGadgets) > 0 {
 		found := false
 
-		for _, registry := range imgOpts.AllowedRegistries {
-			found = strings.HasPrefix(image, registry)
-			if found {
-				break
-			}
-		}
-
-		if !found {
-			return fmt.Errorf("%s not originating from allowed registries: %v", image, strings.Join(imgOpts.AllowedRegistries, ", "))
-		}
-	}
-
-	if len(imgOpts.AllowedDigests) > 0 {
-		targetImage, err := normalizeImageName(image)
+		normalizedImage, err := normalizeImageName(image)
 		if err != nil {
 			return fmt.Errorf("normalizing image: %w", err)
 		}
 
-		desc, err := imageStore.Resolve(ctx, targetImage.String())
+		imageStr := normalizedImage.String()
+
+		desc, err := imageStore.Resolve(ctx, imageStr)
 		if err != nil {
-			return fmt.Errorf("resolving image %q on local registry: %w", targetImage.String(), err)
+			return fmt.Errorf("resolving image %q on local registry: %w", imageStr, err)
 		}
 
-		digestString := desc.Digest.String()
-		found := false
-		for _, digest := range imgOpts.AllowedDigests {
-			if digestString == digest {
-				found = true
+		imageDigest := desc.Digest.String()
+
+		for _, allowedGadget := range imgOpts.AllowedGadgets {
+			if strings.Contains(allowedGadget, "sha") {
+				parts := strings.Split(allowedGadget, ":")
+				if len(parts) != 2 {
+					return fmt.Errorf("%s has badly formed sha", allowedGadget)
+				}
+
+				allowedDigest := parts[1]
+				found = allowedDigest == imageDigest
+			} else if strings.HasSuffix(allowedGadget, "*") {
+				found = strings.HasPrefix(imageStr, allowedGadget[:len(allowedGadget) - 1])
```

Comment:
I'm not sure which one to use `imageStr` or `image`. 

If we use imageStr, then we'll have to use the full qualified name for official-gadgets will be passed, for instance:

```bash
$ sudo -E ig run trace_open:latest --allowed-gadgets=trace_open:latest 
Error: fetching gadget information: initializing and preparing operators: instantiating operator "oci": ensuring image: trace_open:latest is not part of allowed gadgets: trace_open:latest

$ sudo -E ig run trace_open:latest --allowed-gadgets=ghcr.io/inspektor-gadget/gadget/trace_open:latest 
RUNTIME.CONTAINERNA… COMM                PID        TID        UID        GID  FD FNAME                    MODE       ERROR   TIMESTAMP
```

I feel `image` will be less confusing and it doesn't have any bad security implications.